src/scanner.c

```diff
@@ -1265,6 +1275,11 @@ Char GetLine ( void )
 static Char Pushback = '\0';
 static Char *RealIn;
```

Comment:
I realize this is not part of this PR, but: isn't code using these two globals thread unsafe? I.e. if two threads in HPC-GAP were trying to parse a file, they could clash with each other if they both try to "unget" a char simultaneously.

So we should move these into the `GAPState` struct, shouldn't we?

Reply:
That is an *extremely* good comment, and I'm slightly embarassed I didn't think of it. I think I've even seen a couple of crashes about this I could never reproduce (parsing weirdly claiming something couldn't be parsed with weird errors)